TODO.md

```diff
@@ -0,0 +1,30 @@
+# To-do list (temporary)
+
+All items here are subject for discussion.
+
+## Tests
+
+* [x] Interference in nested scopes.
+* [x] Capturing for logger with level set by default.
+* [ ] No race (no records lost) when new records are appended in other thread while being pop-ed.
+* [x] `filter`/`drain` with each filters and combinations, including:
+    * [x] `name` is equal and parent, empty string is equivalent to `None`.
+    * [x] `message` from start, in the middle etc.
+    * [x] Different variants of `exc_info`.
+
+## Features
+
+### Filters
+
+* [x] By `level` (None, int, str), greater or equal (or `level` + `level_exact`? or `level_ge` + `level_eq`?).
+* [x] By `exc_info` (None, bool, class, tuple of classes or tuples).
+* [x] By `stack_info` (optional bool).
+
+### Other
+
+* [x] Better name for `find()`?
+* [ ] `assert_one_pop()` to pop matching and assert only one.  Is it possible to provide custome error message with similar records?   Or full list if it has limitted size.
```

Comment:
I don't get the purpose of this.

Reply:
Most time it's used this way:
```python
recs = pile.drain(…)
assert len(recs) == 1
```
or
```python
recs = pile.drain(…)
assert len(recs) == 1
assert some_cond_for(recs[0])
```
The later can be reduced to somewhat dirty version:
```python
[rec] = pile.drain(…)
assert some_cond_for(rec)
```

But the failure message is not the best in all those cases. The aim is to 1) provide short and clear way to do this, 2) [optionally] provide useful failure report with existing messages so that we don't need to re-run with debugger in case of simple mistakes like typos in name or message pattern.